test/index_selector-spec.ts

```diff
@@ -1,81 +1,110 @@
 
-    it('has a schema and newProcessor method', () => {
-        const processor = indexer;
-
-        expect(processor).toBeDefined();
-        expect(processor.newProcessor).toBeDefined();
-        expect(processor.schema).toBeDefined();
-        expect(typeof processor.newProcessor).toEqual('function');
-        expect(typeof processor.schema).toEqual('function');
+    afterEach(async () => {
+        if (harness) await harness.shutdown();
     });
 
-    it('schema function returns on object, formatted to be used by convict', () => {
-        const schema = indexer.schema();
-        const type = Object.prototype.toString.call(schema);
-        const keys = Object.keys(schema);
+    async function makeTest(_config: any) {
+        const config = Object.assign({}, { _op: 'elasticsearch_index_selector', index: 'some-index', type: 'some-type' }, _config);
+        harness = WorkerTestHarness.testProcessor(config, { assetDir, clients });
+        await harness.initialize();
+        return harness;
+    }
+
+    it('schema has defaults', async () => {
+        const opName = 'elasticsearch_index_selector';
+        const testOpConfig = { _op: opName, index: 'some-index', type: 'some-type' };
+        const test = await makeTest(testOpConfig);
+        const {
+            opConfig: {
+                preserve_id: preserveId,
+                delete: deleteConfig,
+                update,
+                update_retry_on_conflict: retry
+            }
+        } = test.getOperation(opName);
 
-        expect(type).toEqual('[object Object]');
-        expect(keys.length).toBeGreaterThan(0);
-        expect(schema[keys[0]].default).toBeDefined();
+        expect(retry).toEqual(0);
+        expect(preserveId).toEqual(false);
+        expect(deleteConfig).toEqual(false);
+        expect(update).toEqual(false);
     });
 
-    it('new processor will throw if other config options are not present with timeseries', () => {
-        const op1 = { timeseries: 'daily' };
-        const op2 = { timeseries: 'daily', index_prefix: 'hello' };
-        const op3 = { timeseries: 'daily', index_prefix: 'hello', date_field: 'created' };
-
-        expect(() => {
-            indexer.selfValidation(op1);
-        }).toThrowError('elasticsearch_index_selector is mis-configured, if any of the following configurations are set: timeseries, index_prefix or date_field, they must all be used together, please set the missing parameters');
-        expect(() => {
-            indexer.selfValidation(op2);
-        }).toThrowError('elasticsearch_index_selector is mis-configured, if any of the following configurations are set: timeseries, index_prefix or date_field, they must all be used together, please set the missing parameters');
-        expect(() => {
-            indexer.selfValidation(op3);
-        }).not.toThrowError();
-    });
+    it('will throw if other config options are not present with timeseries', async () => {
+        expect.hasAssertions();
```

Comment:
This needs to be the number of assertions that expected to be done. It might be more use `await expect(makeTest(op1)).rejects.toThrowError(new RegExp(`^${errMsg}`));` for rejection tests
```suggestion
        expect.assertions(3);
```